### ISSUE TITLE
Mutating record by assignment to cell paths no longer works in 0.92

### ISSUE THREAD

**Issue by dedebenui** [nushell/nushell#12532]
### Describe the bug

Assignments to cell paths when mutating records raises no error but does nothing.

### How to reproduce

1. Define mutable record
```nushell
mut df = {groups: [a b]} 
```

2. Do some mutating assignment operation
```nushell
$df.groups ++= [c d]  
```

The record is unchanged.

### Expected behavior

Following the steps above, I get the expected result in 0.91.1:
```
╭────────┬───────────╮
│        │ ╭───┬───╮ │
│ groups │ │ 0 │ a │ │
│        │ │ 1 │ b │ │
│        │ │ 2 │ c │ │
│        │ │ 3 │ d │ │
│        │ ╰───┴───╯ │
╰────────┴───────────╯
```

### Configuration

| key                | value                               |
| ------------------ | ----------------------------------- |
| version            | 0.92.2                              |
| branch             |                                     |
| commit_hash        |                                     |
| build_os           | macos-aarch64                       |
| build_target       | aarch64-apple-darwin                |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09) |
| rust_channel       | stable-aarch64-apple-darwin         |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26) |
| build_time         | 2024-04-12 15:51:23 +02:00          |
| build_rust_channel | release                             |
| allocator          | mimalloc                            |
| features           | default, sqlite, trash, which       |
| installed_plugins  |                                     |

**Comment by devyn** [comment]
Hi, we're aware of this issue and it's currently fixed on the `main` branch, which will be released with 0.93.0. The scheduled release date is April 30.

**Comment by dedebenui** [comment]
Ah yes indeed, sorry for the noise then. I should have tried the main branch first 😅
